### PR TITLE
Bootstrap CSS testing ground

### PR DESCRIPTION
**Pull request by hellcp**
So far going on native Bootstrap, so it looks like bland version of current OBS, without green and such. Gonna take care of customization like that later ;)

Should pretty much work anywhere I could test it (in main and package view that is). Scales pretty well except when it doesn't :/

### CODE REVIEW

**Comment by bgeuken** [comment]
We added a switch to our codebase to be able to have some OBS pages rendered by the new framework and rest still using the old bento theme. I would consider using this as well:-)

This would be the commit: https://github.com/openSUSE/open-build-service/pull/4961/commits/de936ce51f821a752115164134623018e35e8e61

https://github.com/openSUSE/open-build-service/pull/4961/commits/c7a2992a5777e32c8af708416ebed5d2f641b933 gives a good example of how this could be used.

**Comment by hellcp** [comment]
Missed it, will look into that :D

File: src/api/config/assets.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Be sure to restart your server when you modify this file.
+
+# Version of your assets, change this if you want to expire all your assets.
+Rails.application.config.assets.version = '1.0'
+Rails.application.config.assets.precompile += ['webui2/application.js', 'webui2/webui2.css.scss']
```
Comment (hennevogel) [inline src/api/config/assets.rb]: I think you need to skip the `.scss` for this to work in production.

**Comment by bgeuken** [comment]
[Decide which framework we want to use to revive the OBS frontend](https://trello.com/c/L0YTakgy/1435-decide-which-framework-we-want-to-use-to-revive-the-obs-frontend)

**Comment by coolo** [comment]
I don't know on what base you plan to decide, but take into consideration:
- software.opensuse.org is bootstrap
- openqa.opensuse.org is bootstrap
- gitlab is bootstrap
- github is bootstrap

So whatever arguments you have contra bootstrap, they must be HEAVY WEIGHT.

**Comment by hennevogel** [comment]
> I don't know on what base you plan to decide

A discussion weighting the pros/cons (but there are not that many) so in the end gut feeling most likely 🤓

> So whatever arguments you have contra bootstrap, they must be HEAVY WEIGHT.

IMHO using a similar stack on openSUSE apps is a thing to consider. But it's not the most important goal I would say. More important for me is that the current community of developers agree that the newly chosen framework will bring the OBS forward _and_ that there is a way for new contributors to get up to speed with it, documentation & community etc.

**Comment by coolo** [comment]
well, where would the new contributors come from? I have little interest to learn yet another css framework.

**Comment by hennevogel** [comment]
> well, where would the new contributors come from?

So far they come from all over the place. Not many have been doing openqa development before ;-)

> I have little interest to learn yet another css framework.

I can totally relate to that. But for others bootstrap would be something they have to learn...

**Comment by hellcp** [comment]
I have heard from Fedora contributor that he would be interested with packaging OBS for Fedora if it was Bootstrap, because Fedora uses Bootstrap a lot too (they are terrible at it, but they try their best I hope).

**Comment by coolo** [comment]
Not many do openqa development in general - different topic though :)

**Comment by coolo** [comment]
For completeness: 
screencapture-localhost-3000-package-show-home-admin-test-2018-05-28-21_43_44

**Comment by hellcp** [comment]
I will have to adapt few Bulma improvements I have seen and haven't implemented, since I didn't want to stray away from original too much. Also build results I didn't do well :D 

Also margins in images, they look kinda weird. Basically WIP :P 

But I feel that what I managed to do so far is nice for few hours thing, so it will be my pleasure to continue working on it (probably tomorrow)

**Comment by Conan-Kudo** [comment]
@coolo Hold on... That looks familiar!

**Comment by coolo** [comment]
that's the beauty of using a framework that everyone else is using. But I only generated the screenshot, it's all @hellcp

**Comment by hennevogel** [comment]
@hellcp are you going to address the feedback?

**Comment by hennevogel** [comment]
@hellcp if you don't have the time I can take over, no biggie :-)

**Comment by hellcp** [comment]
@hennevogel well, I got some troubles with internet connection today, which makes it really hard to setup development environment with Docker ;) Go ahead

**Comment by bgeuken** [comment]
> coolo commented 2 days ago
> well, where would the new contributors come from? I have little interest to learn yet another css framework.

Why did anyone ever started things like containers, new programming languages like Ruby or anything that we have nowadays in IT? How could they know that anyone would want to contribute to these crazy things or use them?
I guess the secret is that one has to try new things out and learn from it.

There might be others that don't want to be involved in yet another app that uses bootstrap and would prefer to get involved with a new framework like bulma or semantic-ui.
We don't know and we can hardly tell:-)

In the end no matter what we pick in the end there will surely be people that totally love what we choose and others that hate it... That's something we have to accept:-)

**Comment by coolo** [comment]
Right. And I didn't say you're idiots to try - I said, the arguments *against* bootstrap need to be HEAVY - because all your peers use bootstrap. 

Trying to rewrite OBS in Go or Haskell because future contributors might like it more is something you wouldn't consider either, right?

**Comment by hellcp** [comment]
@coolo that sounds like FUN, brb in 2 days (jk)

**Comment by Conan-Kudo** [comment]
*(hides in a cave for 10 years to rewrite it all into Python/Rust)*

**Comment by bgeuken** [comment]
> @coolo
> Trying to rewrite OBS in Go or Haskell because future contributors might like it more is something you wouldn't consider either, right?

The difference between those frameworks isn't that drastic. Do you have any example of a contribution that would be considerably harder for a bulma / semantic-ui based OBS than with bootstrap?

**Comment by coolo** [comment]
If the difference isn't drastic, then why even bother with the alternatives?

Even bulma (ignore most of their points against boostrap, they are obsolete with bootstrap 4) names  "Big community" as point for bootstrap: https://bulma.io/alternative-to-bootstrap/

And what I'm saying: SUSE and openSUSE are part of that community (and Fedora, gitlab and github too). Sorry for bulma and semantic UI - I'm sure they are great frameworks and its developers made a good job.

**Comment by bgeuken** [comment]
> If the difference isn't drastic, then why even bother with the alternatives?

That was a reply to your previous polemic suggesting to "rewrite OBS in Go or Haskell":wink:
But yes, for those of us that re-write OBS pages with a new framework it will be some effort. Not as much as rewriting all of OBS, but some.

Once this is done the effort for contributors is hopefully as little as possible. And that is definitely something we consider.
At that time OBS is fully moved to $framework. So contributors, including existing OBS devs, can rely on existing layouts, CSS for certain "objects", and hopefully some re-usable javascript.
In that regard I doubt that the differences in the frameworks we looked at is as big as you make it sound.

**Comment by bgeuken** [comment]
So we (the OBS frontend developers) met today and discussed the pros and cons of the 3 frameworks and decided to use bootstrap :tada:

**Comment by coolo** [comment]
excellent choice!

**Comment by sysrich** [comment]
I am excited!

**Comment by Conan-Kudo** [comment]
Woohoo! Looking forward to it!

**Comment by ChrisBr** [comment]
Superseded in https://github.com/openSUSE/open-build-service/pull/5120

cc @hennevogel

**Comment by hennevogel** [comment]
[Bootstrap: main#index](https://trello.com/c/VqPCDuTh/1451-bootstrap-mainindex)

**Comment by hennevogel** [comment]
[Bootstrap: package#show](https://trello.com/c/VpGsVhbL/1450-bootstrap-packageshow)

**Pull request closed without merging**